scripts/subscribe-logs.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+/*
+Script will create a stream subscription between a log group
+and a lambda function (see arn, below).
+
+ENV vars for ACCOUNT_ID and REGION are required.
+*/
+
+// ACCOUNT_ID=xxxxxxxxx STAGE=omar npm run
+
+const AWS = require('aws-sdk');
+const cloudwatchlogs = new AWS.CloudWatchLogs({ region: 'us-east-1' });
+const accountId = process.env.ACCOUNT_ID;
+const stage = process.env.STAGE;
+const arn = `arn:aws:lambda:us-east-1:${accountId}:function:api-services-${
+  stage
+}-log-streamer`;
+
+const subscribe = item => {
+  let options = {
+    destinationArn: arn,
+    logGroupName: item.logGroupName,
+    filterName: 'stream-logs',
+    filterPattern: '',
+  };
+
+  setTimeout(() => {
+    cloudwatchlogs
+      .putSubscriptionFilter(options)
+      .promise()
+      .then(data => {
+        console.log(data);
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }, 1000);
```

Comment:
Can you please comment on why are you doing thins inside a `setTimeout` call?

Reply:
So as to not exceed AWS API request limits

Reply:
In that case wouldn't you want to schedule it for "a second after the last one"? It seems like you'll simply delay all logGroup subscription calls by a second, but you'll still call them all at once.

Reply:
yes, indeed. I'd meant to multiply it by iteration index.